gptcache/adapter/llama_cpp.py

```diff
@@ -0,0 +1,123 @@
+from typing import Optional, List, Iterator
+import time
+
+from gptcache.adapter.adapter import adapt
+from gptcache.manager.scalar_data.base import DataType, Answer
+
+from gptcache.utils import import_llama_cpp_python
+
+
+import_llama_cpp_python()
+
+import llama_cpp # pylint: disable=wrong-import-position
+
+
+class Llama(llama_cpp.Llama):
+    """llama.cpp wrapper
+
+        You should have the llama-cpp-python library installed.
+        https://github.com/abetlen/llama-cpp-python
+
+    Example:
+        .. code-block:: python
+
+        onnx = Onnx()
+        m = manager_factory('sqlite,faiss,local', data_dir=root, vector_params={"dimension": onnx.dimension})
+        llm_cache = Cache()
+        llm_cache.init(
+            pre_embedding_func=get_prompt,
+            data_manager=m,
+            embedding_func=onnx.to_embeddings
+        )
+        llm = LlamaCpp('./models/7B/ggml-model.bin')
+        answer = llm(prompt=question, cache_obj=llm_cache)
+    """
+    def __call__(
+            self,
+            prompt: str,
+            suffix: Optional[str] = None,
+            max_tokens: int = 128,
+            temperature: float = 0.8,
+            top_p: float = 0.95,
+            logprobs: Optional[int] = None,
+            echo: bool = False,
+            stop: Optional[List[str]] = None,
+            repeat_penalty: float = 1.1,
+            top_k: int = 40,
+            stream: bool = False,
+            **kwargs
```

Comment:
All of the above default parameters can be included in the `kwargs` parameter, like `suffix` , ..., `stream`, like:
```
def __call__(
            self,
            prompt: str,
            **kwargs
```